### PR TITLE
chore: update `@trezor/connect-web` to `v9.2.2`

### PR DESCRIPTION
**Pull request by danjm**
## **Description**

The user experience for trezor connection on mv2 is this:
1. Click through the hardware wallet connect screen for trezor
2. Trezor popup tab opens
3. User can connect the trezor device without having to install other software

Before this PR, on mv3, the flow was:
1. Click through the hardware wallet connect screen for trezor
2. Trezor popup tab opens
3. User has to install the trezor bridge
4. Once installed, the user can connect their trezor device to MetaMask

The solution to the problem required two parts.

First, we needed the `connect-web` module (called `TrezorConnectSDK` in the offscreen trezor file) to specify the `env` variable communicated to the trezor connect iframe and popup to be "webextension" and not just "web". This is because the iframe/popup code assumes that if the iframe origin does not match the origin of popup (e.g. metamask vs connect.trezor.io), and if environment is not "webextension", then webusb cannot be available: https://github.com/trezor/trezor-suite/blob/bb2e075024c8d8316fa016b7b20a0421b1a1f7d0/packages/connect-iframe/src/connectSettings.ts#L74-L84.

In theory, this could be set when the trezor module is initialized, but if the `env` is set to "webextension" in the `connect-web` code, then it attempts to open the popup via chrome apis that are not available in the offscreen document. So the solution is a patch which just tells the iframe and the popup that the environment is "webextension", while leave the environment setting within the trezor code that runs in the offscreen context as "web".

With that fix in place, there was another problem of metamask being able to connect to the webusb device and the popup being able to read that connection (or vice-versa). If the `requestDevice` call happens from the popup, then the webusb permission will be granted to the trezor.io origin, and metamask is unable to communicate with those devices. However, if the `requestDevice` call happens from metamask, then the devices can be communicated from metamask to the popup, and metamask (in particular the offscreen document) can communicate directly with the devices. Specifically, this code https://github.com/trezor/trezor-suite/blob/bb2e075024c8d8316fa016b7b20a0421b1a1f7d0/packages/transport/src/api/usb.ts#L98, when run within the offscreen document, returns null when `requestDevice` was called from the trezor popup, but returns a correctly populated array if `requestDevice` was called from MetaMask. To fix this, we can call `requestDevice` explicitly within the MetaMask ui flow, instead of having the trezor popup call it for us

Finally, and independent of the above two fixes, this PR updates trezor connect packages, which was necessary to get trezor working properly with Snow.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23763?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Build this branch with `yarn start:mv3`
2. Go through the regular trezor connect flow
3. Trezor should connect and sign as normal

## **Screenshots/Recordings**

https://github.com/MetaMask/metamask-extension/assets/7499938/29533595-f16f-4492-be21-f9bc3698fe70

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by brad-decker — DISMISSED** [review]

File: development/build/scripts.js [hunk 1]
```diff
@@ -1047,6 +1047,13 @@ async function createBundle(buildConfiguration, { reloadOnChange }) {
   // forward update event (used by watchify)
   bundler.on('update', () => performBundle());
 
+  // trezor requires `ts-mixer`, but it pulls in the ESM version for some
+  // reason. ESM doesn't work with our build system.. uh, because ESM sucks and
+  // I hope the author lives a life of pain and agony, but I digress. This
+  // work around makes browserify (which also sucks) always use the CJS version.
+  // How? Magic (I don't know).
+  bundler.require(require.resolve('ts-mixer'), { expose: 'ts-mixer' });
```
Comment (davidmurdoch) [inline development/build/scripts.js:1055]: this breaks the build for some websites (tested on facebook and instagram). I'll need to fix this by patching `ts-mixer` to always return it's CJS version.

File: ui/pages/create-account/connect-hardware/select-hardware.js [hunk 2]
```diff
@@ -51,10 +51,28 @@ export default class SelectHardware extends Component {
...
+              { vendorId: 0x534c, productId: 0x0001 },
+              { vendorId: 0x1209, productId: 0x53c0 },
+              { vendorId: 0x1209, productId: 0x53c1 },
+            ],
+          });
+          this.setState({ trezorRequestDevicePending: false });
+        } catch (e) {
+          this.setState({ trezorRequestDevicePending: false });
+          throw e;
+        }
```
Comment (davidmurdoch) [inline ui/pages/create-account/connect-hardware/select-hardware.js:73]: nite: something like the following is slightly cleaner

```suggestion
        } catch (e) {
          throw e;
        } finally {
          this.setState({ trezorRequestDevicePending: false });
        }
```

**Comment by danjm** [comment]
@metamaskbot update-policies

File: ui/pages/create-account/connect-hardware/select-hardware.js (hunk 2 above)
Comment (DDDDDanica) [reply ui/pages/create-account/connect-hardware/select-hardware.js]: Addressed !

File: test/e2e/tests/account/import-flow.spec.js [hunk 3]
```diff
@@ -377,7 +377,7 @@ describe('Import flow @no-mmi', function () {
     );
   });
 
-  it('Connects to a Hardware wallet for trezor', async function () {
+  it('Connects to a Hardware wallet for lattice', async function () {
     if (process.env.ENABLE_MV3) {
```
Comment (legobeat) [inline test/e2e/tests/account/import-flow.spec.js:381]: It looks like this replaces the existing test for Trezor with a test for GridPlus Lattice. Is there a reason to not keep testing for the Trezor-connect case?

File: test/e2e/tests/account/import-flow.spec.js [hunk 4]
```diff
@@ -401,11 +401,16 @@ describe('Import flow @no-mmi', function () {
           text: 'Add hardware wallet',
           tag: 'button',
         });
-        await driver.delay(regularDelayMs);
-
+        await driver.findClickableElement(
+          '[data-testid="hardware-connect-close-btn"]',
+        );
         // should open the TREZOR Connect popup
```
Comment (legobeat) [inline test/e2e/tests/account/import-flow.spec.js:407]:
```suggestion
```

**Review by DDDDDanica — DISMISSED** [review]

**Comment by DDDDDanica** [comment]
The implementation looks good to me. Approved ~

**Comment by danjm** [comment]
@SocketSecurity ignore-all

All of the dependencies that socket security is warning us about are dependencies of trezor, but not of them are used in our applications code

**Review by DDDDDanica — APPROVED** [review]

**Review by vthomas13 — APPROVED** [review]

**Pull request merged**